### ISSUE TITLE
Sanitize input should not cache previous sanitize values

### ISSUE THREAD

**Issue by aaronbhansen** [DockYard/ember-one-way-controls#103]
I tried created a create a sanitizedInput where only numbers should be accepted on a phone number input. 

If all you enter is invalid results, the first characters is removed because of the sanitize function, but since the sanitize function returns empty string after that, the update method is never called again because the previous sanitized result is cached. 

Once you enter a valid character, the update method is called and the invalid characters are removed. See example https://cl.ly/2o0x442f0V0I

Feels like the sanitizedValue shouldn't be cached https://github.com/DockYard/ember-one-way-controls/blob/master/addon/components/one-way-input.js#L78

or specifically check that a sanitize action was passed in or a sanitize method exists on the component so you don't have to cache the result each time.

**Comment by martndemus** [comment]
Can you show me some sample code that you have used to produce this issue?

**Comment by aaronbhansen** [comment]
Template
```
{{one-way-tel
    address.phone_number
    id="phone_number"
    placeholder="+54 341 555 5555"
    flags="i"
    tabindex="7"
    autocomplete="tel"
    name="telephone"
    update=(pipe-action (action (mut address.phone_number)) (action "changed"))
    sanitizeInput=(action "sanitizePhoneNumber")}}
```

component action
```
sanitizePhoneNumber(phone_number) {
  if (phone_number == null) {
    return phone_number;
  }
  return phone_number.replace(/[^0-9\\ \(\)\+\-\.\#\*\~\/ext]/gi, '');
}
```

**Comment by martndemus** [comment]
Can you test if it now works with 1.0.1?

**Comment by aaronbhansen** [comment]
I still can type "aaaaa" and it isn't sanitizing the input since after the first character the sanitization results is the same as the first time and it's cached.

**Comment by martndemus** [comment]
@aaronbhansen Can you try if the following works for you?

```hbs
{{one-way-tel
    address.phone_number
    id="phone_number"
    placeholder="+54 341 555 5555"
    flags="i"
    tabindex="7"
    autocomplete="tel"
    name="telephone"
    update=(pipe-action (action "sanitizePhoneNumber") (action (mut address.phone_number)) (action "changed"))}}
```

**Comment by aaronbhansen** [comment]
It does work, but I was hoping to use sanitize before mutating the value so that the value update action only got called if the value changed. That was one of the big reasons for using one-way-controls was that the update method only got called if something changed.

**Comment by martndemus** [comment]
This should now be fixed in `1.0.2`
